File: cpp/src/evaluator.cc
/*
 *  Copyright 2016-2019 Henry Lee
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

#include "phevaluator/phevaluator.h"

namespace phevaluator {

int EvaluateCards(Card a, Card b, Card c, Card d, Card e) {
  return evaluate_5cards(a, b, c, d, e);
}

int EvaluateCards(Card a, Card b, Card c, Card d, Card e, Card f) {
  return evaluate_6cards(a, b, c, d, e, f);
}

int EvaluateCards(Card a, Card b, Card c, Card d, Card e, Card f, Card g) {
  return evaluate_7cards(a, b, c, d, e, f, g);
}

int EvaluateCards(Card a, Card b, Card c, Card d, Card e, Card f, Card g, Card h) {
  return evaluate_8cards(a, b, c, d, e, f, g, h);
}

int EvaluateCards(Card a, Card b, Card c, Card d, Card e, Card f, Card g, Card h, Card i) {
  return evaluate_9cards(a, b, c, d, e, f, g, h, i);
}

} // namespace phevaluator

File: python/phevaluator/evaluator/evaluator.py
from evaluator.card import Card
from evaluator.evaluator5 import evaluate_5cards
from evaluator.evaluator6 import evaluate_6cards
from evaluator.evaluator7 import evaluate_7cards
from evaluator.evaluator8 import evaluate_8cards
from evaluator.evaluator9 import evaluate_9cards

def evaluate_cards(*args):
  if len(args) == 5:
    return evaluate_5cards(
      Card(args[0]), Card(args[1]), Card(args[2]), Card(args[3]), Card(args[4])
    )
  elif len(args) == 6:
    return evaluate_6cards(
      Card(args[0]), Card(args[1]), Card(args[2]),
      Card(args[3]), Card(args[4]), Card(args[5])
    )
  elif len(args) == 7:
    return evaluate_7cards(
      Card(args[0]), Card(args[1]), Card(args[2]), Card(args[3]),
      Card(args[4]), Card(args[5]), Card(args[6])
    )
  elif len(args) == 8:
    return evaluate_8cards(
      Card(args[0]), Card(args[1]), Card(args[2]), Card(args[3]),
      Card(args[4]), Card(args[5]), Card(args[6]), Card(args[7])
    )
  elif len(args) == 9:
    return evaluate_9cards(
      Card(args[0]), Card(args[1]), Card(args[2]), Card(args[3]), Card(args[4]),
      Card(args[5]), Card(args[6]), Card(args[7]), Card(args[8])
    )
